src/AppInstallerRepositoryCore/Microsoft/Schema/Portable_1_0/PortableTable.cpp

```diff
@@ -0,0 +1,160 @@
+        SQLite::Builder::StatementBuilder builder;
+        builder.Select(SQLite::RowIDName).From(s_PortableTable_Table_Name).Where(s_PortableTable_FilePath_Column);
+        builder.Equals(path.u8string());
+
+        SQLite::Statement select = builder.Prepare(connection);
+
+        if (select.Step())
+        {
+            return select.GetColumn<SQLite::rowid_t>(0);
+        }
+        else
+        {
+            return {};
+        }
+    }
+
+    void PortableTable::RemovePortableFileById(SQLite::Connection& connection, SQLite::rowid_t id)
+    {
+        SQLite::Builder::StatementBuilder builder;
+        builder.DeleteFrom(s_PortableTable_Table_Name).Where(SQLite::RowIDName).Equals(id);
+        builder.Execute(connection);
+    }
+
+    SQLite::rowid_t PortableTable::AddPortableFile(SQLite::Connection& connection, const PortableFile& file)
+    {
+        SQLite::Builder::StatementBuilder builder;
+        builder.InsertInto(s_PortableTable_Table_Name)
+            .Columns({ s_PortableTable_FilePath_Column,
+                s_PortableTable_FileType_Column,
+                s_PortableTable_SHA256_Column,
+                s_PortableTable_SymlinkTarget_Column,
+                s_PortableTable_IsCreated_Column })
+            .Values(file.FilePath, file.FileType, file.SHA256, file.SymlinkTarget, file.IsCreated);
+
+        builder.Execute(connection);
+        return connection.GetLastInsertRowID();
+    }
+
+    bool PortableTable::UpdatePortableFileById(SQLite::Connection& connection, SQLite::rowid_t id, const PortableFile& file)
+    {
+        SQLite::Builder::StatementBuilder builder;
+        builder.Update(s_PortableTable_Table_Name).Set()
+            .Column(s_PortableTable_FilePath_Column).Equals(file.FilePath)
+            .Column(s_PortableTable_FileType_Column).Equals(file.FileType)
+            .Column(s_PortableTable_SHA256_Column).Equals(file.SHA256)
+            .Column(s_PortableTable_SymlinkTarget_Column).Equals(file.SymlinkTarget)
+            .Column(s_PortableTable_IsCreated_Column).Equals(file.IsCreated)
+            .Where(SQLite::RowIDName).Equals(id);
+
+        builder.Execute(connection);
+        return connection.GetChanges() != 0;
+    }
+
+    std::optional<PortableFile> PortableTable::GetPortableFileById(const SQLite::Connection& connection, SQLite::rowid_t id)
+    {
+        SQLite::Builder::StatementBuilder builder;
+        builder.Select({ s_PortableTable_FilePath_Column,
+            s_PortableTable_FileType_Column,
+            s_PortableTable_SHA256_Column,
+            s_PortableTable_SymlinkTarget_Column,
+            s_PortableTable_IsCreated_Column })
+            .From(s_PortableTable_Table_Name).Where(SQLite::RowIDName).Equals(id);
+
+        SQLite::Statement select = builder.Prepare(connection);
+
+        PortableFile portableFile;
+        if (select.Step())
+        {
+            portableFile.FilePath = select.GetColumn<std::string>(static_cast<int>(PortableColumnType::FilePath));

```

Comment:
This is fragile; the index of the column in the result will match the order that they are in the select statement.

If you don't even want to care about index numbers, you should be able to do:
```
[portableFile.FilePath, ...] = select.GetRow<std::string, ...>();
```
Where you replace the `...`s with the appropriately ordered `portableFile` fields and types. respectively.